Pchain UTs cleanup

abi87 opened the pull request:
## Why this should be merged
platformVM unit tests needs to be easier to maintain.
Specifically I currently find non-trivial to understand what specific fork a unit test is testing and possibly to update the tests to the latest fork.
Also fork times do not make sense and vm clock is not set to fullfil them.

## How this works
Just refactor tests so that the latest active fork is explictly set upon vm creation and vm.clock is set to make the fork active.

## How this was tested
Testing test is tricky. I guess I tested by manual inspection

anusha-ctrl commented on vms/platformvm/block/builder/helpers_test.go:
```diff
@@ -303,16 +348,17 @@ func defaultConfig() *config.Config {
 	clk := &mockable.Clock{}
-	clk.Set(defaultGenesisTime)
+	clk.Set(now)
 	return clk
```
is it cleaner to just say `clk.Set(latestForkTime)`

anusha-ctrl replied:
```suggestion
func defaultClock() *mockable.Clock {
clock := &mockable.Clock{}
clock.Set(latestForkTime)
return clock
}
```

anusha-ctrl commented on vms/platformvm/block/executor/verifier_test.go:
```diff
@@ -29,10 +29,15 @@ import (
 
+	config := &config.Config{
+		BanffTime:   mockable.MaxTime, // banff is not activated
+		CortinaTime: mockable.MaxTime, // cortina is not activate
```
```suggestion
		CortinaTime: mockable.MaxTime, // cortina is not activated
```

anusha-ctrl replied:
same for below

anusha-ctrl replied:
it might be worth creating a `notActivatedConfig`

anusha-ctrl commented on vms/platformvm/state/staker_test.go:
```diff
@@ -143,8 +143,8 @@ func TestNewCurrentStaker(t *testing.T) {
-	endTime := time.Now()
+	startTime := time.Now().Truncate(time.Second)
+	endTime := time.Now().Truncate(time.Second)
 	potentialReward := uint64(54321)
```
Why was this changed?

abi87 commented:
Replaced by https://github.com/ava-labs/avalanchego/pull/2169. Thanks @anusha-ctrl for the comments that pushed me to revise this PR (and replaced it).
